src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -285,6 +304,9 @@ export const DataTable: FunctionComponent<Props> = ({
       return <div>FEATURE FLAG IN USE...</div>;
     }
 
+    // TODO (mlila): this is the source of constant rerendering in the table. Because the rows are
+    // TODO          rendered with a function instead of a react component, it generates a new node
+    // TODO          every time the parent rerenders, instead of only updating, eg, when props change
```

Comment:
should we create a ticket to address this? Seems like this should be refactored to be a component

Reply:
Possibly, but tbh it feels like part of the table refactor epic. The problem is that FixedSizeList expects the child to be shaped this way. I haven't spent any time playing with it to see if we could use a component instead, but it may be an artifact we have to deal with until we use another table library.